src/query-builder/RelationLoader.ts

```diff
@@ -191,6 +191,9 @@ export class RelationLoader {
 
         Object.defineProperty(entity, relation.propertyName, {
             get: function() {
+                if (this[dataIndex])
```

Comment:
its not a proper fix since `this[dataIndex]` can be lets say `0` or lets say a `Promise`. That's why there is a `resolveIndex` that really indicates if there is a data or not.

So, the question is - howcome do we have dataIndex set but resolveIndex unset? Probably investigations should go to find out why setter is not being called?

Reply:
Maybe can help to give me clue on this, 
```RelationLoader``` is not being triggered, thats why ```ret``` is still undefined.

on ```EntityMetadata -> create``` this error shows up.

```
"SyntaxError: Unexpected end of input
    at EntityMetadata.create (/Users/markmatute/Desktop/final-ts-koa/server/src/metadata/EntityMetadata.ts:495:13)
    at RawSqlResultsToEntityTransformer.transformRawResultsGroup (/Users/markmatute/Desktop/final-ts-koa/server/src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts:90:100)
    at /Users/markmatute/Desktop/final-ts-koa/server/src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts:44:33
    at Map.forEach (<anonymous>)
    at RawSqlResultsToEntityTransformer.transform (/Users/markmatute/Desktop/final-ts-koa/server/src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts:43:15)
    at SelectQueryBuilder.<anonymous> (/Users/markmatute/Desktop/final-ts-koa/server/src/query-builder/SelectQueryBuilder.ts:1826:36)
    at step (/Users/markmatute/Desktop/final-ts-koa/server/node_modules/typeorm/query-builder/SelectQueryBuilder.js:42:23)
    at Object.next (/Users/markmatute/Desktop/final-ts-koa/server/node_modules/typeorm/query-builder/SelectQueryBuilder.js:23:53)
    at fulfilled (/Users/markmatute/Desktop/final-ts-koa/server/node_modules/typeorm/query-builder/SelectQueryBuilder.js:14:58)
    at <anonymous>"
```